lib/prototyp/bigint.go

```diff
@@ -228,29 +246,58 @@ func (b *BigInt) UnmarshalBinary(buff []byte) error {
 
 func ParseNumberString(s string, base int) (int64, bool) {
 	var ns strings.Builder
-	if base == 10 {
+	switch base {
+	case 2:
+		for _, char := range s {
+			// 0-9 || B || b
+			if (char >= 48 && char <= 57) || char == 45 || char == 66 || char == 98 {
+				ns.WriteRune(char)
```

Comment:
We should, but it is written in a way, that it ignores other characters.
So `1a2b3c4d` is parsed to `1234`
I can't change it now, it would be breaking change :/

Reply:
I think it would be for better. Ignoring some characters and treating the input as valid sounds like a bug to me.

Also,  I think we should introduce a function that returns an error.
```go
func ParseNumberString() (int64, error) {}
```

but to keep this one backward-compatible, we'd need to come up with a different name

@pkieltyka thoughts?

Reply:
we definitely need backwards compatibility, as well what would we use `func ParseNumberString() (int64, error) {}` for..? if anything we should have: `ParseNumberStringInt64` and `ParseNumberStringBigInt` for example.. I think we can do those in a separate PR, lets keep this PR focused to get the bug fixed